Model/Sync/Orders/Data.php

```diff
@@ -144,12 +144,23 @@ public function prepareData($order, $syncType, $yotpoOrderObject)
         $shippingAddress = $order->getShippingAddress();
         $orderStatus = $order->getStatus();
         $mappedYotpoOrderStatus = $this->getYotpoOrderStatus($orderStatus);
+        $payment = $order->getPayment();
+        if ($payment === null) {
+            $this->yotpoOrdersLogger->info(
+                __('Order with no payment object - Order ID: %1, Increment ID: %2',
+                    $order->getEntityId(),
+                    $order->getIncrementId()
+                )
+            );
+            return null;
```

Comment:
Should we return null and avoid the order itself or just not mentioning the payment?